conferences/ticketing_processing.go

```diff
@@ -0,0 +1,145 @@
+	"context"
+	"fmt"
+
+	"encore.dev/storage/sqldb"
+	uuid "github.com/satori/go.uuid"
+)
+
+// claimSlots claims N slots for an attendee.
+func claimSlots(ctx context.Context, attendee *Attendee, slots ...ConferenceSlot) ([]SlotClaim, error) {
+	tx, err := sqldb.Begin(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("beginning atomic operation: %w", err)
+	}
+	var claims = make([]SlotClaim, 0, len(slots))
+	for i := range slots {
+		slot := slots[i]
+		sc := &SlotClaim{
+			ConferenceSlot: &slot,
+			TicketID:       uuid.NewV4().String(),
+		}
+		sc, err = createSlotClaim(ctx, tx, sc)
+		if err != nil {
+			if atomicErr := sqldb.Rollback(tx); atomicErr != nil {
+				err = fmt.Errorf("%w (also cancelling atomic operation: %v)", err, atomicErr)
+			}
+			return nil, fmt.Errorf("Claiming a slot: %w", err)
+		}
+		claims[i] = *sc
+	}
+	attendee.Claims = append(attendee.Claims, claims...)
+	_, err = updateAttendee(ctx, tx, attendee)
+	if err != nil {
+		if atomicErr := sqldb.Rollback(tx); atomicErr != nil {
+			err = fmt.Errorf("%w (also cancelling atomic operation: %v)", err, atomicErr)
+		}
+		return nil, fmt.Errorf("Updating claimed slots for attendee: %w", err)
+	}
+	if err := sqldb.Commit(tx); err != nil {
+		return nil, fmt.Errorf("confirming atomic operation: %w", err)
+	}
+	return claims, nil
+}
+
+// payClaims assigns payments and/or credits to a set of claims.
+func payClaims(ctx context.Context, attendee *Attendee, claims []SlotClaim,
+	payments []FinancialInstrument) (*ClaimPayment, error) {
+	ptrClaims := make([]*SlotClaim, len(claims))
+	for i := range claims {
+		ptrClaims[i] = &claims[i]
+	}
+	claimPayment := &ClaimPayment{
+		ClaimsPayed: ptrClaims,
+		Payment:     payments,
+	}
+	tx, err := sqldb.Begin(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("beginning atomic operation: %w", err)
+	}
+
+	claimPayment, err = createClaimPayment(ctx, tx, claimPayment)
+	if err != nil {
+		if atomicErr := sqldb.Rollback(tx); atomicErr != nil {
+			err = fmt.Errorf("%w (also cancelling atomic operation: %v)", err, atomicErr)
+		}
+		return nil, fmt.Errorf("paying for claims: %w", err)
+	}
+	if err := sqldb.Commit(tx); err != nil {
+		return nil, fmt.Errorf("confirming atomic operation: %w", err)
+	}
+
+	return claimPayment, nil
+}
+
+// ErrInvalidCurrency should be returned when paying with the wrong kind of instrument
+// for instance covering credit with credit.
+type ErrInvalidCurrency struct {
+	currencyType AssetType
+}
+
+func (e *ErrInvalidCurrency) Error() string {
+	return fmt.Sprintf("the debt cannot be covered with %s", e.currencyType)
+}
+
+// CoverCredit adds funds to a payment to cover for receivables.
+func coverCredit(ctx context.Context,
+	existingPayment *ClaimPayment,
+	payments []FinancialInstrument) error {
+	for i := range payments {
+		if payments[i].Type() == ATReceivable {
+			return &ErrInvalidCurrency{currencyType: payments[i].Type()}
+		}
+		existingPayment.Payment = append(existingPayment.Payment, payments[i])
+	}
```

Comment:
We can make this nicer by assigning the value to a variable. 

```suggestion
	for i, payment := range payments {
		if payment.Type() == ATReceivable {
			return &ErrInvalidCurrency{currencyType: payment.Type()}
		}
		existingPayment.Payment = append(existingPayment.Payment, payment)
	}
```